src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -31,11 +32,32 @@ impl ScalarFunc {
         let res = t.date_format(format_mask_str)?;
         Ok(Some(Cow::Owned(res.into_bytes())))
     }
+
+    #[inline]
+    pub fn date<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
+        let t = try_opt!(self.children[0].eval_time(ctx, row));
+        if t.invalid_zero() {
+            return Err(box_err!("Incorrect datetime value: '{}'", t));
+        }
+        if t.is_zero() {
+            return Err(box_err!("Incorrect datetime value: '{}'", t));
```

Comment:
Could we define a similar error with the same error code and the similar message like tidb?